library/dnet.c

```diff
@@ -1013,6 +1484,9 @@ static int dnet_process_cmd_with_backend_raw(struct dnet_backend_io *backend, st
 	gettimeofday(&end, NULL);
 	diff = DIFF(start, end);
 
+	/* If there was any error - send ACK to notify client with error code and destroy transaction */
+	if (err)
+		cmd->flags |= DNET_FLAGS_NEED_ACK;
```

Comment:
Maybe they do, that would be a bug.

This particular patch fixes timeouts for write errors with compare-and-swap and failed checksums, since write never sends any reply in this case. This is also true for all other commands which do not explicitly set NEED_ACK bit.